draft-irtf-cfrg-vdaf.md

```diff
@@ -1190,6 +1190,202 @@ how the nonces are chosen, but security requires that the nonces be unique. See
 execution of a VDAF requires the application to instantiate secure channels
 between each of the protocol participants.
 
+## Communication Patterns for Preparation {#vdaf-prep-comm}
+
+In each round of preparation, each Aggregator writes a prep share to the
+channel, which is then processed into the prep message using the public
+`prep_shares_to_prep` algorithm and broadcast to the Aggregators to start the
+next round. In this section we describe some approaches to realizing this
+broadcast channel functionality in protocols that use VDAFs.
+
+The state machine of each Aggregator for VDAF preparation is shown in
+{{vdaf-prep-state-machine}}.
+
+~~~ state
+               +----------------+
+               |                |
+               v                |
+Start ------> Continued(prep_state) --> Finished(out_share)
+ |                |
+ |                |
+ +--> Rejected <--+
+~~~
+{: #vdaf-prep-state-machine title="State machine for VDAF preparation."}
+
+State transitions are made when the state is acted upon by the host's local
+inputs and/or messages sent by the peers. The initial state is `Start`. The
+terminal states are `Rejected`, indicating that the report was rejected and
+cannot be processed any further; and `Finished(out_share)` where `out_share` is
+the Aggreagtor's recovered output share.
+
+## Ping-Pong Topology (Only Two Aggregators)
+
+For VDAFs with precisely two Aggregators (i.e., `Vdaf.SHARES == 2`), the
+following "ping pong" communication pattern can be used. It is compatible with
+any request/response protocol, such as HTTP.
+
+The first state transition, from `Start` to `Continued` or `Rejected`, is
+induced by the following transition rule. No messages are sent or received
+during this transition.
+
+~~~ transition
+def ping_pong_start(Vdaf,
+                    vdaf_verify_key: bytes[Vdaf.VERIFY_KEY_SIZE],
+                    is_leader: bool,
+                    agg_param: Vdaf.AggParam,
+                    nonce: bytes[Vdaf.NONCE_SIZE],
+                    public_share: bytes,
+                    host_input_share: bytes) -> State:
+    try:
+        host_prep_state = Vdaf.prep_init(
+            vdaf_verify_key,
+            0 if is_leader else 1,
+            agg_param,
+            nonce,
+            public_share,
+            host_input_share,
+        )
+        return Continue(host_prep_state)
```

Comment:
At the risk of being a pedant: if the VDAF being run was 0-round, then `Vdaf.prep_init` would transition right to `Finished`. I suppose you wouldn't bother running the ping-pong mode for a 0-round VDAF at all, but I feel like either this section could state that it's only defined for VDAFs with at least one round, or `{{vdaf}}` might say that `VDAF.ROUNDS > 0`.